### ISSUE TITLE
Kafka-Goblin on MR - Error, even if Topic is written on HDFS

### ISSUE THREAD

**Issue by loryfelnunez** [linkedin/gobblin#652]
Running map-reduce on Kafka seem to consume the topic from my Kafka producer and write to HDFS.  But I get the errors below.

WARN [AbstractJobLauncher] Not committing dataset  of job job_GobblinKafkaQuickStart_1454370378013 with commit policy COMMIT_ON_FULL_SUCCESS and state FAILED
ERROR [AbstractJobLauncher] Failed to launch and run job job_GobblinKafkaQuickStart_1454370378013: java.io.IOException: Failed to commit dataset state for some dataset(s) of job job_GobblinKafkaQuickStart_1454370378013

Failed to launch the job due to the following exception:
gobblin.runtime.JobException: Job job_GobblinKafkaQuickStart_1454370378013 failed
WARN [ContextAwareReporter] Reporter MetricReportReporter has already been stopped.
WARN [ContextAwareReporter] Reporter MetricReportReporter has already been stopped.

When I look at the my Hadoop, I see Diagnostics: Call From ip-xxxx to 0.0.0.0:9000 failed on connection exception: java.net.ConnectException: 

When I do 
yarn logs -applicationId application_1454278218699_0009
16/02/02 01:09:29 INFO client.RMProxy: Connecting to ResourceManager at ec2-xxxxx:8050
/tmp/logs/ubuntu/logs/application_1454278218699_0009 does not exist.
Log aggregation has not completed or is not enabled.

Help?

**Comment by sahilTakiar** [comment]
Hey @loryfelnunez the logs are basically saying that one of the Map tasks failed. Unfortunately, to get the full stack trace of the Map Task you have to read the Map Task Logs.

Can you try `yarn logs -applicationId application_1454278218699_0009` again? Sometimes its takes a while for the logs to get aggregated.

Alternatively, do you have Job History store running? This should allow you the view the logs of the Map Tasks via a UI.

**Comment by loryfelnunez** [comment]
Thanks Sahil.

I looked at my Job History and saw that my Hadoop configuration had an errant
<property>
    <name>fs.default.name</name>
    <value>hdfs://0.0.0.0:9000</value>
  </property>
In my NameNode.  

I removed it and my job seems to go further, until I hit this error:
Error: java.lang.ClassNotFoundException: org.apache.commons.lang3.StringUtils

I added to map-reduce.sh
LIBJARS=$FWDIR_LIB/commons-lang3-3.4.jar
and downloaded and added
$FWDIR_LIB/typesafe-config-2.10.1.jar
 
WARN [MRJobLauncher] Job working directory already exists for job GobblinKafkaQuickStart
Error: java.io.IOException: Not all tasks running in container attempt_1454278218699_0017_m_000002_0 completed successfully
	at gobblin.runtime.AbstractJobLauncher.runWorkUnits(AbstractJobLauncher.java:580)

My Hadoop version is 2.7.1

I went to http://ec2xxx.compute-1.amazonaws.com:8088/cluster/appattempt/appattempt_xxxxx

Diagnostics Info:	
Task failed task_1454278218699_0017_m_000003
Task failed task_1454278218699_0017_m_000002

I will be looking at how to get the logs of this failed job.  Seems like my yarn log aggregation is not enabled..
